server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -161,14 +181,14 @@ public DimensionFilter getRangeMatchFilter(
         Long parsedLow = rawLow == null ? defaultMinimum() : numberFieldType.numberType().parse(rawLow, true).longValue();
         Long parsedHigh = rawHigh == null ? defaultMaximum() : numberFieldType.numberType().parse(rawHigh, true).longValue();
 
-        boolean lowerTermHasDecimalPart = hasDecimalPart(parsedLow);
+        boolean lowerTermHasDecimalPart = hasDecimalPart(rawLow);
```

Comment:
If we are going to fix this bug in this PR, let's write a UT for the same OR not fix in this PR.

Reply:
Looks like another PR was opened for this. https://github.com/opensearch-project/OpenSearch/pull/17372 

Since we should encourage new contributors, let's fix this bug in that PR.